From c3c70e87b040233263b9594d14582dfedfecc92e Mon Sep 17 00:00:00 2001
From: Yaroslav Kholod <y.kholod@vyos.io>
Date: Wed, 18 Dec 2024 11:48:29 +0200
Subject: [PATCH] #17413: Clean babeld config on stop

---
 babeld/babeld.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/babeld/babeld.c b/babeld/babeld.c
index b562f0b70..6f1a9a3d7 100644
--- a/babeld/babeld.c
+++ b/babeld/babeld.c
@@ -304,6 +304,12 @@ void babel_clean_routing_process(void)
     flush_all_routes();
     babel_interface_close_all();
 
+    /* Clean babel config */
+    diversity_kind = DIVERSITY_NONE;
+    diversity_factor = BABEL_DEFAULT_DIVERSITY_FACTOR;
+    resend_delay = BABEL_DEFAULT_RESEND_DELAY;
+    smoothing_half_life = BABEL_DEFAULT_SMOOTHING_HALF_LIFE;
+
     /* cancel events */
     event_cancel(&babel_routing_process->t_read);
     event_cancel(&babel_routing_process->t_update);
-- 
2.43.0

